### PR TITLE
Add tests for [Sign In] Issue 62, 63, 81

### PR DESCRIPTION
**Pull request by drozdovskyy**

### CODE REVIEW

File: src/main/java/com/academy/ui/components/HeaderComponent.java [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
+
+    @FindBy(how = How.CSS, using = "li.user-name")
+    protected WebElement profileLink;
+
+    @FindBy(how = How.CSS, using = ".header_sign-up-link")
+    protected WebElement signInComponentRoot;
+
     public HeaderComponent(WebDriver driver, WebElement rootElement) {
         super(driver, rootElement);
+        PageFactory.initElements(driver,this);
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/HeaderComponent.java]: remove this;

File: src/main/java/com/academy/ui/components/HeaderComponent.java [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+    }
+
+    public SignInComponent getSignInComponent() {
+        return new SignInComponent(driver, signInComponentRoot);
     }
 
+    public void clickSignInLink() {
+        signInLink.click();
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/HeaderComponent.java:40]: return new SignInComponent(driver, signInComponentRoot);

File: src/main/java/com/academy/ui/components/HeaderComponent.java [hunk 3]
```diff
@@ -1,26 +1,52 @@
...
+    protected WebElement signInComponentRoot;
+
     public HeaderComponent(WebDriver driver, WebElement rootElement) {
         super(driver, rootElement);
+        PageFactory.initElements(driver,this);
+    }
+
+    public SignInComponent getSignInComponent() {
+        return new SignInComponent(driver, signInComponentRoot);
     }
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/HeaderComponent.java]: remove

File: src/main/java/com/academy/ui/components/SignInComponent.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    @FindBy(how = How.ID, using = "password")
+    protected WebElement passwordInput;
+
+    @FindBy(how = How.CSS, using = "button.greenStyle")
+    protected WebElement signInButton;
+
+
+    public SignInComponent(WebDriver driver, WebElement rootElement) {
+        super(driver, rootElement);
+        PageFactory.initElements(driver, this);
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/SignInComponent.java]: remove this

File: src/main/java/com/academy/ui/components/SignInComponent.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+
+    // Functional
+
+    private void enterEmailInput(String email) {
+        clickEmailInput();
+        clearEmailInput();
+        setEmailInput(email);
+    }
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/SignInComponent.java]: private SignInComponent enterEmailInput(String email) {
        clickEmailInput();
        clearEmailInput();
        setEmailInput(email);
        return this;
    }

File: src/main/java/com/academy/ui/components/SignInComponent.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    public SignInComponent fillEmailInput(String email) {
+        enterEmailInput(email);
+        return this;
+    }
+
+    public SignInComponent fillPasswordInput(String password) {
+        enterPasswordInput(password);
+        return this;
+    }
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/SignInComponent.java]: remove

File: src/main/java/com/academy/ui/components/SignInComponent.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    public HomePage successfulSignIn(String email, String password) {
+        fillLogin(email, password);
+        clickSignInButton();
+        return new HomePage(driver);
+    }
+
+    public SignInComponent unsuccessfulSignIn() {
+        return this;
+    }
+}
```
Comment (lhalam) [inline src/main/java/com/academy/ui/components/SignInComponent.java]: public void sendSignInForm() {
        clickSignInButton();
    }

    public HomePage successfulSignIn() {
         sendSignInForm()
        return new HomePage(driver);
    }

    public SignInComponent unsuccessfulSignIn() {
        sendSignInForm()
        return this;
    }

**Comment by drozdovskyy** [comment]
slightly changed solution

```
   public SignInComponent clickSignInButton() {
        signInButton.click();
        return this;
    }

    public HomePage successfulSignIn() {
        return new HomePage(driver);
    }
```

File: src/main/java/com/academy/ui/components/SignInComponent.java (hunk 6 above)
Comment (drozdovskyy) [reply src/main/java/com/academy/ui/components/SignInComponent.java]: done

File: src/main/java/com/academy/ui/components/SignInComponent.java (hunk 5 above)
Comment (drozdovskyy) [reply src/main/java/com/academy/ui/components/SignInComponent.java]: done

File: src/main/java/com/academy/ui/components/SignInComponent.java (hunk 4 above)
Comment (drozdovskyy) [reply src/main/java/com/academy/ui/components/SignInComponent.java]: done

**Comment by drozdovskyy** [comment]
All changes have been implemented

**Pull request closed without merging**